pactsdk/pool.py

```diff
@@ -1,123 +1,88 @@
-    CONFIG: Any = None
-
-
-@dataclass
-class PoolState:
-    total_liquidity: int
-    total_primary: int
-    total_secondary: int
-    primary_asset_price: D
-    secondary_asset_price: D
-
-
 def list_pools(pact_api_url: str, **params):
     assert pact_api_url
     encoded_params = urlencode(params)
     response = requests.get(f"{pact_api_url}/api/pools?{encoded_params}")
     return response.json()
 
 
-def fetch_app_state(
+def fetch_app_global_state(
     algod: AlgodClient,
     app_id: int,
 ) -> AppInternalState:
     app_info = algod.application_info(app_id)
-    state_dict = parse_global_state(app_info["params"]["global-state"])
-    return AppInternalState(**state_dict)
-
-
-def parse_global_state(kv: list) -> dict[str, int]:
-    # Transform algorand key-value schema into python dict with key value pairs
-    res = {}
-    for elem in kv:
-        key = str(base64.b64decode(elem["key"]), encoding="ascii")
-        if elem["value"]["type"] == 1:
-            val = elem["value"]["bytes"]
-        else:
-            val = elem["value"]["uint"]
-        res[key] = val
-    return res
-
-
-def fetch_pool(
-    algod: AlgodClient,
-    asset_a: Asset,
-    asset_b: Asset,
-    app_id: int = 0,
-    fee_bps: int = 30,
-    pact_api_url="",
-) -> "Pool":
-    # Make sure that the user didn't mess up assets order.
-    # Primary asset always has lower index.
-    primary_asset, secondary_asset = sorted([asset_a, asset_b], key=lambda a: a.index)
+    return parse_global_pool_state(app_info["params"]["global-state"])
 
-    if not app_id:
-        assert pact_api_url, "Must provide pactifyApiUrl or app_id."
 
-        app_id = get_app_id_from_assets(
-            pact_api_url,
-            primary_asset,
-            secondary_asset,
-        )
-        if not app_id:
-            raise PactSdkError(
-                f"Cannot find pool for assets {primary_asset.index} and {secondary_asset.index}.",
-            )
+def fetch_pool_by_id(algod: AlgodClient, app_id: int):
+    app_global_state = fetch_app_global_state(algod, app_id)
 
-    app_state = fetch_app_state(algod, app_id)
-    liquidity_asset = fetch_asset_by_index(algod, app_state.LTID)
+    primary_asset = fetch_asset_by_index(algod, app_global_state.ASSET_A)
+    secondary_asset = fetch_asset_by_index(algod, app_global_state.ASSET_B)
+    liquidity_asset = fetch_asset_by_index(algod, app_global_state.LTID)
 
     return Pool(
         algod=algod,
         app_id=app_id,
         primary_asset=primary_asset,
         secondary_asset=secondary_asset,
         liquidity_asset=liquidity_asset,
-        internal_state=app_state,
-        fee_bps=fee_bps,
+        internal_state=app_global_state,
+        fee_bps=app_global_state.FEE_BPS,
     )
 
 
-def get_app_id_from_assets(
+def fetch_pools_by_assets(
+    algod: AlgodClient,
+    asset_a: Union[Asset, int],
+    asset_b: Union[Asset, int],
+    pact_api_url="",
```

Comment:
pact_api_url: str=""

Reply:
not needed, mypy will interpolate type based on literal value